portfolio/src/main/webapp/script.js

```diff
@@ -26,3 +26,37 @@ function addRandomGreeting() {
   const greetingContainer = document.getElementById('greeting-container');
   greetingContainer.innerText = greeting;
 }
+
+/**
+ * Adds a random fact to the page.
+ */
+function addRandomFact() {
+    const facts = ["I have no sibling", 
+                   "I seldom eat spicy food", 
+                   "I don't like watching thrillers",
+                   "I love drinking milk tea"];
+
+    // Pick a random fact.
+    const factContainer = document.getElementById("fact-container");
+    let currentFact = factContainer.innerText;
+    let fact = getRandomFact(facts, currentFact);
+
+    // Add it to the page.
+    factContainer.innerText = fact;
+} 
+
+/**
+ * Picks a fun fact to be displayed.
+ * @param {string array} facts A list of facts.
+ * @param {string} currentFact Current fun fact.
+ * @return {string} Picked fun fact.
+ */
+function getRandomFact(facts, currentFact) {
+    let factIndex = Math.floor(Math.random() * facts.length)
+    let fact = facts[factIndex];
+    while (fact == currentFact) {
```

Comment:
in stead of while loop, can we compute the next index by (current index + some random) mod length?

Reply:
Fixed